### ISSUE TITLE
Training table read of year_built_data omits fallbackByReference

### ISSUE THREAD

**Issue by ZiolkowskiJakub** [ZiolkowskiJakub/DiGi.GIS.ML#7]
> Found during a full production audit of `year_built_data` on 2026-09-09 against
> `https://api.digiproject.uk` (build `0.8.8.20260909083309`). The table itself is **clean**:
> 410 712 rows over 17 counties, **0 orphans**, **0 duplicate rows**, and nothing filed under a
> foreign partition. The findings below are about how it could stop being clean.

## Defect

`YearBuiltPredictionTrainingTableConsoleApp/Query/YearBuiltDatasAsync.cs` reads `year_built_data`
pruned to one county and **without** `fallbackByReference` (`:45` builds the reference list, `:97`
sets only `countyid`).

`year_built_data` is partitioned `BY LIST (county_id)`, and the read prunes on it. A row filed under
a **sibling polygon part** of the same county code is therefore not returned, and the building drops
out of the training table with no error - it simply looks like a building that carries no label.

## Why that is worse here than elsewhere

This is the **training set**. A silently missing label is not a missing row in a report; it changes
what the regressor learns, and nothing downstream can tell that it happened. The counties that
carry `UserYearBuilt` labels are exactly the small ones this audit found
(`0201` 10 440, `0202` 3 799, `2264` 3 640, `2476` 4 338 rows), so a proportionally small loss is
a large fraction of the ground truth.

## The inconsistency

The pipeline that *writes* these rows sets the flag deliberately, and says why:

```csharp
// DiGi.GIS.YOLO.UI/Query/YearBuiltDatasAsync.cs:74-78
// ... without the flag a stored row filed under a sibling polygon part is no longer read back,
// which is what strands a duplicate.
```

The equivalent omission in `PostgreSQLBuildingDataUpdateTask` is filed separately against
`DiGi.GIS.PostgreSQL`.

## Impact today

None measurable - the audit found **0 rows under a wrong part** across 410 712 rows, and none of
the four label counties is a multi-part code. Latent, and worth closing before a multi-part county
is ever labelled or scored.

## Suggested fix

Set `fallbackbyreference=true` on the `itemsbyreferences` request. Keep `CancellationToken` last
(CA1068) and pass the flag by name.

**Comment by ZiolkowskiJakub** [comment]
## Resolution & commits

- **Repository:** `DiGi.GIS.ML`
- **Branch:** `0.8.8` (version branch), pushed to `origin/0.8.8`
- **Commit:** [`0bfc0ab`](https://github.com/ZiolkowskiJakub/DiGi.GIS.ML/commit/0bfc0ab67da082429ec6a408ea62d6f726e91bd4)

## Summary of changes

The `YearBuiltDatasAsync` training-table read built the `itemsbyreferences` request URL with only `countyid`, so `fallbackbyreference` fell back to the server default of `false`. With the flag off, the read prunes to the single county part it was given, so a `YearBuiltData` row filed under a sibling polygon part of a multi-part county was silently dropped from the training set. This is the training set, so a dropped label is not a missing row - it is a building the regressor never sees, and nothing downstream can tell that it happened.

The items read now sets the flag explicitly:

- `YearBuiltPredictionTrainingTableConsoleApp/Query/YearBuiltDatasAsync.cs` - the items read appends `AddParameter("fallbackbyreference", true)`, with a comment and a summary `<para>` explaining the stakes.
- `documentation/API/YearBuiltPredictionTrainingTableConsoleApp/DiGi.GIS.ML.ConsoleApp.md` - the generated API doc reflects the new `<para>`.

**Wire-name note.** The parameter is sent as lowercase `fallbackbyreference` (value `true`, serialized to `True` by `UrlBuilder`) to match the controller binding `[FromQuery(Name = "fallbackbyreference")]` and the two existing production call sites that already rely on it (the `DiGi.GIS.YOLO.UI` writing pipeline and the `DiGi.GIS.WebAPI` client). Query parameter binding is case-sensitive, so the name must match the declared wire name exactly.

## Automated & integration tests

No new `[Fact]` was added: this read lives in a console app that the `DiGi.GIS.ML.xUnit` test project does not reference (it references the library DLLs only), the Deployed-WebAPI guideline keeps live-API reads as manual curl rather than `DiGi.Test`, and the guard is preventive - there is no reproducible defect on the current data to pin an assertion to.

Verification:

- `dotnet build YearBuiltPredictionTrainingTableConsoleApp/YearBuiltPredictionTrainingTableConsoleApp.csproj -c Debug` -> 0 warnings, 0 errors.
- Cross-checked the sent query against the controller binding and both production call sites - the request now matches the working production request in parameter name and value.

## Live deployed verification

Read-only check against `api.digiproject.uk` on county `104106` (a single-part county): the endpoint accepts `fallbackbyreference` (no `400`) and returns the full reference set. On a single-part county the flag is a no-op, so both values return the identical set; the multi-part case, where the flag actually changes the result, was not exercised live. The binding guarantee comes from matching, in name and value, the exact request the working production call sites send to the same `itemsbyreferences` endpoint.
